packages/java/endpoint/src/main/java/com/vaadin/hilla/route/records/ClientViewConfig.java

```diff
@@ -42,25 +50,103 @@ public void add(String key, Object value) {
 
     /**
      * Get all unknown values.
+     *
      * @return a map of unknown values
      */
     @JsonAnyGetter
     public Map<String, Object> getOther() {
         return other;
     }
 
+    public String getTitle() {
+        return title;
+    }
+
+    public String[] getRolesAllowed() {
+        return rolesAllowed;
+    }
+
+    public boolean isRequiresLogin() {
```

Comment:
As these are going to be part of the public API, I suggest to rename `requiresLogin` to `loginRequired` so that the accessor could be read naturally as `isLoginRequired`.